### PR TITLE
Implement debug utility for elastio-snap

### LINKED ISSUE #272: Create a skeleton of the driver debug utility
In order to address requests from the customers and debug the driver more effectively, we need a debugging tool

1. We need to understand better what’s going on with the driver
2. Customers/support engineers need to understand better how to configure it
3. At some point we’ll need a tool to provide remote support for a Customer

The main metrics to achieve:

 - What information is written and where it’s mapped in the COW file?
 - How many free blocks have we still got there?
 - How many disk IO per second take place?
 - What is an average device throughput?
 - How much RAM is used by the driver in real time?
 - What is the bio request latency? Are there bottlenecks anywhere?
 - Any errors during the operation?

For the implementation, Netlink sockets are proposed. The debug information will be streamed locally, where the proxy utility translates it to the usable form for the further redirect. For the nearest time and internal debugging, the simple architecture is proposed:

elastio-snap -> proxy -> GUI application

The goal is to create the skeleton, estimate its pros and cons, and decide if it fits our needs.

### PR DESCRIPTION
**Pull request by skypodolsky**
This commit introduces the console debug utility for the elastio-snap driver. The design is the following:
 - driver sends Netlink events
 - the events are read by `nl_debug` (debugging utility)
 - events are proxied by using the UDP protocol
 - parsed and printed in the console if needed

The events are sent as multicast messages, so potential packet drops are possible. All messages have sequence numbers and `nl_debug` detects if any packet drops took place. If yes, it specifies the amount of missed packets. Despite the drops are possible, practically, they don't occur.

The utility supports different filtering that simplifies debugging:
 - bio in the sector range
 - bio of the specific type (read/write)
 - muting of different events by groups (COW file & bio events)
 - all messages are colored depending on the type to simplify reading

The Netlink part is not compiled by default, so there is a 0% chance it can degrade the driver performance. A special flag should be set manually to include it:

`sudo make NETLINK_DEBUG=y`

Closes #272

### CODE REVIEW

File: src/nl_debug.h [hunk 1]
```diff
@@ -0,0 +1,107 @@
+#ifdef KERNEL_MODULE
```
Comment (e-kov) [inline src/nl_debug.h:1]: Missing license and copyright headers. Let's use `LGPL-2.1-or-later` if `nl_debug` can be used as library in the future?
```
/* SPDX-License-Identifier: LGPL-2.1-or-later */

/*
 * Copyright (C) 2023 Elastio Software Inc.
 */
```

File: src/nl_debug.h [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+#define trace_event_cow(_type, _priv1, _priv2)	\
+({ 												\
+	struct params_t params = { 0 }; 			\
+												\
+	params.priv1 = (_priv1); 					\
+	params.priv2 = (_priv2); 					\
+	nl_send_event(_type, __func__, __LINE__, &params); \
+})
+
```
Comment (e-kov) [inline src/nl_debug.h:104]: It seems backslashes `\` can be better aligned =)

Comment (skypodolsky) [reply src/nl_debug.h:111]: It just because of the tabs, it looks nice in the code

File: src/nl_debug.h (hunk 1 above)
Comment (skypodolsky) [reply src/nl_debug.h:8]: Done!

**Review by e-kov — APPROVED** [review]

**Review by vsazhenyuk-softheme — APPROVED** [review]

**Comment by skypodolsky** [comment]
> That said, why is UDP used here? Is there a scenario where we need to run the driver on one host and collect metrics on another?

We might need some proxy in the future (this utility was planned as a proxy initially), so I decided to lay some foundation for that 🙂

**Pull request merged**